release-content/0.16/release-notes/gpu_driven_rendering.md

```diff
@@ -0,0 +1,75 @@
+## GPU-Driven Rendering
+
+Over the years, the trend in real-time rendering has increasingly been to move work from the CPU to the GPU. One of the latest developments in this area has been *GPU-driven rendering*, in which the GPU takes a representation of the scene and essentially works out what to draw on its own. While Bevy 0.15 had some support for GPU-driven rendering for meshlets only, Bevy 0.16 features comprehensive support for this technique, including for skinned meshes. This dramatically reduces the amount of CPU time that the renderer needs for larger scenes. It's automatically enabled on platforms that support it; unless your application hooks into the rendering pipeline, upgrading to Bevy 0.16 will automatically enable GPU-driven rendering for your meshes.
+
+To explain how GPU-driven rendering operates, it's easiest to first describe how CPU-driven rendering works:
+
+1. The CPU determines the objects that are visible, via frustum culling and perhaps occlusion culling.
+
+2. For each such object:
+
+   a. The CPU sends the object's transform to the GPU, possibly in addition to other data such as joint weights.
+
+   b. The CPU tells the GPU where the mesh data is.
+
+   c. The CPU writes the material data to the GPU.
+
+   d. The CPU tells the GPU where textures and other buffers needed to render the objects are (light data, etc.)
+
+   e. The CPU issues a drawcall.
+
+   f. The GPU renders the object.
+
+In contrast, GPU-driven rendering in Bevy works like this:
+
+1. The CPU uploads transform information for all objects at once to the GPU.
+
+2. If new objects have been spawned since the last frame, the CPU fills out tables specifying where the mesh data for the new objects are.
+
+3. If materials have been modified since the last frame, the CPU uploads information about those materials to the GPU.
```

Comment:
This is the same as for CPU-driven.

Reply:
Yeah, I feel like the _real_ GPU-driven part is the culling. Stuff like bindless, multidraw, and caching out specialization are all important, and could be covered in separate section, but are not really GPU-driven techniques (I suppose you could argue bindless is).

Reply:
My view is that "GPU driven" is a combination of many techniques, none of which *on their own* indicate "GPU driven", but when you put them all together they add up to that technique. GPU driven isn't culling, bindless, multidraw, or cached specialization: it's what happens when you *combine* all of those.